acp/acp_source_hub.go

```diff
@@ -0,0 +1,288 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package acp
+
+import (
+	"context"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	protoTypes "github.com/cosmos/gogoproto/types"
+	"github.com/sourcenetwork/immutable"
+	"github.com/sourcenetwork/sourcehub/sdk"
+	acptypes "github.com/sourcenetwork/sourcehub/x/acp/types"
+
+	"github.com/sourcenetwork/defradb/acp/identity"
+	"github.com/sourcenetwork/defradb/keyring"
+)
+
+type acpSourceHub struct {
```

Comment:
> we should likely then name sourcehubClient to coreACPClient

`sourceHubClient` has nothing to do with acp-core, and I would not be in favour of that name until/unless it does.  Same goes for `coreACPBridge`

Reply:
Do you see local acp (which only uses core acp), as a sourceHubClient?

Reply:
It does implement the interface, and is consumed by the bridge.  I think you are confused as to what these interfaces are.

Reply:
> where local acp used to be local sourcehub acp

It was never really "local sourcehub ACP" though, that was a technical coincendence since the "embedded" ACP which local ACP used, was implemented inside the sourcehub repo, but didn't *use* SourceHub in that sense.